### DISCUSSION TITLE
Persistent Windows not working as well with Nvidia card vs Integrated Graphics

### DISCUSSION THREAD

**Discussion by CalculonPrime** [kangyu-california/PersistentWindows#248] [General]
With Intel Integrated Graphics, I had the problem of windows moving around when monitors got disconnected, which Persistent Windows fixed.  However, after switching to an Nvidia card, I'm seeing a new problem. In addition to the windows moving around, they're also resized smaller - Persistent windows resizes them back, but it's too late.  For example some of the windows are VMWare machines, and the windows inside of the VM get squished down due to this happening.  The VM window getting resized back to its original size doesn't unsquish the windows inside the VM.  This didn't happen before with Intel Graphics.

Is there any known solution to this?  I don't have a Quadro card with built-in EDID management.  Someone recommended a software called CRU.  Not sure which problem that solves - the same one as Persistent Windows, or whether it's able to stop the resizing completely.

**Comment by kangyu-california** [comment]
I have some idea to fix, but still need more details to confirm. Could you take two screen shots (before and after restore), picture is worth a lot of words

**Comment by CalculonPrime** [comment]
OK, here is a screenshot from before I switch monitors, and I after I switch back.  You can see that the VMWare window has been resized properly, but the explorer window inside has been squished, indicating that the host machine (on which Persistent Windows is running) did at one point resize the VMWare window smaller, squishing the contents.

The weird thing is that this didn't happen when I was using Intel Integrated Graphics.  It's a new problem after having installed an Nvidia graphics card and driver.
![vmware1](https://github.com/kangyu-california/PersistentWindows/assets/57440041/e22c283d-424b-4548-86d4-c92380f25019)
![vmware2](https://github.com/kangyu-california/PersistentWindows/assets/57440041/e334b555-6863-4b5d-9b60-912bce9ff296)

**Reply by kangyu-california** [reply]
Each vm also needs to install persistentwindows

**Reply by CalculonPrime** [reply]
Dear God, no!  I've got VMs running Windows XP, as well as Linux.  I'm not going to install software into all of them.  There has to be some other solution.  Why is Nvidia doing this when the integrated graphics didn't?

Have you heard about the CRU software?

**Reply by kangyu-california** [reply]
Instead of figuring out why this video card or monitor works while the other doesn't and prevent undesirable window move from happening in the first place, PersistentWindows (and a bunch of similar software tools like MonitorKeeper/WindowResizer/WindowLayoutSnapshot/winredock) admits that the "why" issue is too complicated to generalize/prevent and focus on correct the window move in a hardware agnostic way.

**Comment by CalculonPrime** [comment]
Should I create a separate thread to ask about the CRU software? I assume that's something you're familiar with as it seems to be related to the issue Persistent Windows is solving, though I'm not really sure.  There's a forum for it, but the discussion thread has 761 pages so it's not something easily digested.

**Reply by kangyu-california** [reply]
You can share the experience of using CRU here. But I am not be able to answer questions on CRU.

**Reply by CalculonPrime** [reply]
I was just looking for, broad picture, what is does, and whether it solves the same problem as Persistent Windows.

**Reply by kangyu-california** [reply]
It sounds like asking a windows XP question at a linux forum

**Reply by kangyu-california** [reply]
![image](https://github.com/kangyu-california/PersistentWindows/assets/59128756/2c133502-0a02-4c66-ab07-0e51cc0bf1cf)
